### PR TITLE
[#4185] improvement(core, catalogs): Change the @VisibleForTesting in production code to change usability

### PR DESCRIPTION
**Pull request by khmgobe**
### What changes were proposed in this pull request?

Modifying declaration parts, removing calls that do not need to be used

### Why are the changes needed? It shouldn't be used, but it's declared

Fix: #4185 

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Check change code and existing code comparison

### CODE REVIEW

**Comment by khmgobe** [comment]
@justinmclean As you said, I posted PR after reorganization. Thank you for your advice.

File: catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java [hunk 1]
```diff
@@ -41,9 +41,9 @@
 public class PaimonCatalogPropertiesMetadata extends BaseCatalogPropertiesMetadata {
 
   @VisibleForTesting public static final String GRAVITINO_CATALOG_BACKEND = "catalog-backend";
-  @VisibleForTesting public static final String PAIMON_METASTORE = "metastore";
-  @VisibleForTesting public static final String WAREHOUSE = "warehouse";
-  @VisibleForTesting public static final String URI = "uri";
+  public static final String PAIMON_METASTORE = "metastore";
+  public static final String WAREHOUSE = "warehouse";
+  public static final String URI = "uri";
```
Comment (jerryshao) [inline catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:46]: My comment from previous PR, can we make these variables be package public, I don't think they need to be public if they're only for tests.

Comment (justinmclean) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:46]: My understanding is that they are used outside of tests.

Comment (khmgobe) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:46]: > My comment from previous PR, can we make these variables be package public, I don't think they need to be public if they're only for tests.

@jerryshao 

I respect your opinion, but it was already used outside. What are your opinions on this?

Comment (jerryshao) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:46]: I see, I'm fine with this.

Comment (khmgobe) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:46]: @jerryshao Thank you for your constructive comments.

Comment (khmgobe) [reply catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/PaimonCatalogPropertiesMetadata.java:46]: @justinmclean Did you have a good day today? If you have time, please check.

File: catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java [hunk 2]
```diff
@@ -235,7 +234,6 @@ private void initKerberosIfNecessary(Map<String, String> conf, Configuration had
 
         LOG.info("krb5 path: {}", System.getProperty("java.security.krb5.conf"));
         refreshKerberosConfig();
-        KerberosName.resetDefaultRealm();
```
Comment (yuqi1129) [inline catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java:238]: ` KerberosName.resetDefaultRealm();` has no real effect in the current context, we can freely remove it.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Comment by khmgobe** [comment]
> LGTM

thanks!

**Pull request merged**